runtime-light/stdlib/rpc/rpc-api.cpp

```diff
@@ -0,0 +1,522 @@
+    co_return error_factory.make_error(string{"rpc query has empty result fetcher"}, TL_ERROR_INTERNAL);
+  }
+  if (!rpc_query.get()->result_fetcher->is_typed) {
+    co_return error_factory.make_error(string{"can't get typed result from untyped TL query. Use consistent API for that"}, TL_ERROR_INTERNAL);
+  }
+
+  const auto data{co_await f$component_client_get_result(component_query)};
+
+  // TODO: subscribe to rpc response event?
+  // update rpc response extra info
+  if (const auto it_response_extra_info{rpc_ctx.rpc_responses_extra_info.find(query_id)}; it_response_extra_info != rpc_ctx.rpc_responses_extra_info.end()) {
+    const auto timestamp{std::chrono::duration<double>{std::chrono::system_clock::now().time_since_epoch()}.count()};
+    it_response_extra_info->second.second = {data.size(), timestamp - std::get<1>(it_response_extra_info->second.second)};
+    it_response_extra_info->second.first = rpc_response_extra_info_status_t::READY;
+  }
+
+  rpc_ctx.fetch_state.reset(0, data.size());
+  rpc_ctx.buffer.clean();
+  rpc_ctx.buffer.append(data.c_str(), data.size());
+
+  co_return fetch_function_typed(rpc_query, error_factory);
+}
+
+} // namespace rpc_impl_
+
+// === Rpc Store ==================================================================================
+
+bool f$store_int(int64_t v) noexcept {
+  if (unlikely(is_int32_overflow(v))) {
+    php_warning("Got int32 overflow on storing '%" PRIi64 "', the value will be casted to '%d'", v, static_cast<int32_t>(v));
+  }
+  return rpc_impl_::store_trivial(static_cast<int32_t>(v));
+}
+
+bool f$store_long(int64_t v) noexcept {
+  return rpc_impl_::store_trivial(v);
+}
+
+bool f$store_float(double v) noexcept {
+  return rpc_impl_::store_trivial(static_cast<float>(v));
+}
+
+bool f$store_double(double v) noexcept {
+  return rpc_impl_::store_trivial(v);
+}
+
+bool f$store_string(const string &v) noexcept { // TODO: support large strings
+  auto &buffer{RpcComponentContext::get().buffer};
+
+  string::size_type string_len{v.size()};
+  string::size_type size_len{};
+  if (string_len <= rpc_impl_::SMALL_STRING_MAX_LEN) {
+    buffer << static_cast<uint8_t>(string_len);
+    size_len = 1;
+  } else if (string_len <= rpc_impl_::MEDIUM_STRING_MAX_LEN) {
+    buffer << static_cast<uint8_t>(rpc_impl_::MEDIUM_STRING_MAGIC) << static_cast<uint8_t>(string_len & 0xff) << static_cast<uint8_t>((string_len >> 8) & 0xff)
+           << static_cast<uint8_t>((string_len >> 16) & 0xff);
+    size_len = 4;
+  } else {
+    buffer << static_cast<uint8_t>(rpc_impl_::LARGE_STRING_MAGIC) << static_cast<uint8_t>(string_len & 0xff) << static_cast<uint8_t>((string_len >> 8) & 0xff)
+           << static_cast<uint8_t>((string_len >> 16) & 0xff) << static_cast<uint8_t>((string_len >> 24) & 0xff);
+//    buffer << static_cast<uint8_t>(rpc_impl_::LARGE_STRING_MAGIC) << static_cast<uint8_t>(string_len & 0xff) << static_cast<uint8_t>((string_len >> 8) & 0xff)
```

Comment:
Why this comment here? What does it mean?

Reply:
I'm gonna uncomment it as soon as we support large strings in TL